vspace/src/main/webapp/WEB-INF/views/staff/spaces/space.jsp

```diff
@@ -294,6 +300,109 @@ $( document ).ready(function() {
+	            $("#bgImage").off("click");
+	            moduleLinkInfo["id"] = linkData["id"];
+	            showModuleLinkEdit(editModuleLinkInfo, true);
+	            $("#editModuleLinkInfo").hide();
+	            $("#errorMsg").text("");
+	            $('#errorAlert').hide();
+	        }
+	    });
+	});
+	
+	$("#editSpaceLinkBtn").click(function(e) {
+	    e.preventDefault();
+	    var linkId = $("#spaceLinkIdEdit").val();
+	    if (storeX == undefined || storeY == undefined) {
+	        $("#errorMsg").text("Please click on the image to specify where the new link should be located.")
+	        $('#errorAlert').show();
+	        return;    
+	    }
+	    var linkedSpace = $("#spaceLinkIdEdit").val();
+	    $("#spaceLinkXEdit").val(storeX);
+	    $("#spaceLinkYEdit").val(storeY);
+
+	    var form = $("#editSpaceLinkForm");
+	    var formData = new FormData(form[0]);
+	    var spaceLinkInfo = editSpaceLinkInfo();
+	    $.ajax({
+	        type: "POST",
+	        url: "<c:url value="/staff/space/link/space/${space.id}?${_csrf.parameterName}=${_csrf.token}" />",
+	        cache: false,
+	        contentType : false,
+	        processData : false,
+	        enctype: 'multipart/form-data',
+	        data: formData,
+	        success: function(data) {
+	            var linkData = JSON.parse(data);
+	            $("#bgImage").off("click");
+	            spaceLinkInfo["id"] = linkData["id"];
+	            showSpaceLinkEdit(editSpaceLinkInfo, true);
+	            $("#editSpaceLinkInfo").hide();
+	            $("#errorMsg").text("");
+	            $('#errorAlert').hide();
+	       	}
+	 	});
+	});
+	
+	$("#editExternalLinkBtn").click(function(e) {
+	    e.preventDefault();
+	    var linkId = $("#externalLinkIdEdit").val();
+	    if (storeX == undefined || storeY == undefined) {
+	        $("#errorMsg").text("Please click on the image to specify where the new link should be located.")
+	        $('#errorAlert').show();
+	        return;
+	    }
+	    var linkedSpace = $("#externalLinkIdEdit").val();
+	    $("#externalLinkXEdit").val(storeX);
+	    $("#externalLinkYEdit").val(storeY);
+
+	    var form = $("#editExternalLinkForm");
+	    var formData = new FormData(form[0]);
+	    var externalLinkInfo = editExternalLinkInfo();
+	    $.ajax({
+	        type: "POST",
+	        url: "<c:url value="/staff/space/link/external/${space.id}?${_csrf.parameterName}=${_csrf.token}" />",
+	        cache: false,
+	        contentType : false,
+	        processData : false,
+	        enctype: 'multipart/form-data',
+	        data: formData,
+	        success: function(data) {
+	            var linkData = JSON.parse(data);
+	            $("#bgImage").off("click");
+	            externalLinkInfo["id"] = linkData["id"];
+	            showExternalLinkEdit(editExternalLinkInfo, true);
+	            $("#editExternalLinkInfo").hide();
+	            $("#errorMsg").text("");
+	            $('#errorAlert').hide();
+	       	}
+	 	});
+	});
```

Comment:
this looks like it's the same code for all three link types (with just different html elements, classes ,etc). common code should be in methods so you don't repeat yourself.